test/e2e_node/shortened_grace_period_test.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2enode
+
+import (
+	"context"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+var _ = framework.KubeDescribe("ShortenedGracePeriod [NodeFeature:ShortenedGracePeriod]", func() {
+	f := framework.NewDefaultFramework("shortened-grace-period")
+	ginkgo.Context("When repeatedly deleting pods", func() {
+		ginkgo.It("should be deleted immediately", func() {
+			nodeName := getNodeName(f)
+			podName := "test"
+
+			var gracePeriod int64 = 20
+			f.PodClient().CreateSync(getGracePeriodTestPod(podName, nodeName, gracePeriod))
+			err := f.PodClient().Delete(context.TODO(), podName, metav1.DeleteOptions{
+				GracePeriodSeconds: &gracePeriod,
+			})
+			framework.ExpectNoError(err)
+
+			start := time.Now()
+			var gracePeriodZero int64 = 0
+			f.PodClient().DeleteSync(podName, metav1.DeleteOptions{
+				GracePeriodSeconds: &gracePeriodZero,
+			}, 1*time.Second)
+			framework.ExpectEqual(time.Since(start) < 3*time.Second, true, "kubelet cannot update GracePeriodSeconds")
```

Comment:
the `kubelet` isnt really the actor here, right?  the error message is slightly confusing as a result.